synapse/handlers/e2e_keys.py

```diff
@@ -977,12 +985,66 @@ def _get_e2e_cross_signing_verify_key(self, user_id, key_type, from_user_id=None
         Raises:
             NotFoundError: if the key is not found
         """
+        user = UserID.from_string(user_id)
         key = yield self.store.get_e2e_cross_signing_key(
             user_id, key_type, from_user_id
         )
+
+        if key is None and self.is_mine(user):
+            # Attempt to fetch the missing key from the remote user's server
```

Comment:
this is quite a big if statement. Can you factor out its body please?

Reply:
I've changed it up a bit. Do you still think it's too big?

Reply:
sorry, I missed this. hrm, it's a bit marginal. up to you I guess.